pkg/operator/resource/resourceapply/storage.go

```diff
@@ -173,31 +174,57 @@ func ApplyCSIDriver(ctx context.Context, client storageclientv1.CSIDriversGetter
 		}
 	}
 
-	metadataModified := false
+	needsUpdate := false
+	// Most CSIDriver fields are immutable. Any change to them should trigger Delete() + Create() calls.
+	needsRecreate := false
+
 	existingCopy := existing.DeepCopy()
-	resourcemerge.EnsureObjectMeta(&metadataModified, &existingCopy.ObjectMeta, required.ObjectMeta)
+	// Metadata change should need just Update() call.
+	resourcemerge.EnsureObjectMeta(&needsUpdate, &existingCopy.ObjectMeta, required.ObjectMeta)
 
 	requiredSpecHash := required.Annotations[specHashAnnotation]
 	existingSpecHash := existing.Annotations[specHashAnnotation]
```

Comment:
So here if we recomputed hash using `existing` object, it won't match the `required` object's hash. Should we do that instead? It is not like computing hash of `CSIDriver` spec is expensive or anything?

And we could do that all the time IMO? Changing the value IIRC doesn't require kubelet or external-attacher restart iirc

Reply:
> because of defaulting. E.g spec.requiresRepublish in our operators is usually nil, but the API server defaults it to false and such comparison would detect a difference.

That seems fixable, but okay.  Lets consider this this comment as resolved. Do you want to add some tests?

Reply:
Good catch, we need a test. I'll check if it's possible and how ugly it would be.